### PR TITLE
feat(transport): Add buffer size option to native transport

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [x] Bugfix
- [x] New feature
- [x] Enhancement
- [x] Refactoring

## :scroll: Description
iOS Doesn't have an option for maxQueueSize, therefore the option is only passed to Java and JS layers.

## :bulb: Motivation and Context
Fixes: https://github.com/getsentry/sentry-react-native/issues/2497

## :green_heart: How did you test it?
Sample app.

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+### Features
+- Add buffer size option to native transport ([#2578](https://github.com/getsentry/sentry-react-native/pull/2578))
```
Comment (marandaneto) [inline CHANGELOG.md:6]: H: Always jump a line :)

File: android/src/main/java/io/sentry/react/RNSentryModule.java [hunk 2]
```diff
@@ -130,6 +130,12 @@ public void initNativeSdk(final ReadableMap rnOptions, Promise promise) {
             if (rnOptions.hasKey("sendDefaultPii")) {
                 options.setSendDefaultPii(rnOptions.getBoolean("sendDefaultPii"));
             }
+            if (rnOptions.hasKey("transportOptions")) {
+                final ReadableMap transportOptions = rnOptions.getMap("transportOptions");
+                if (transportOptions.hasKey("bufferSize")) {
+                    options.setMaxQueueSize(rnOptions.getInt("bufferSize"));
+                }
+            }
```
Comment (marandaneto) [inline android/src/main/java/io/sentry/react/RNSentryModule.java:138]: H: This should be called `maxQueueSize` and it should be a high-level property, not under `transportOptions`.

File: src/js/sdk.tsx [hunk 3]
```diff
@@ -40,6 +40,7 @@ const DEFAULT_OPTIONS: ReactNativeOptions = {
   enableOutOfMemoryTracking: true,
   patchGlobalPromise: true,
   transportOptions: {
+    bufferSize: DEFAULT_BUFFER_SIZE,
```
Comment (marandaneto) [inline src/js/sdk.tsx:43]: It's ok to call it `bufferSize` here since it inherits from JS.
But for RN Apps, lets stick to `maxQueueSize` high level property, so one only does `options.maxQueueSize = X`.

Comment (marandaneto) [reply src/js/sdk.tsx:43]: Relates to https://github.com/getsentry/sentry-react-native/pull/2578/files#r1007056274

Comment (krystofwoldrich) [reply src/js/sdk.tsx:43]: I can add `maxQueueSize` but then we will be advertising two places where people can set the `bufferSize`/`maxQueueSize`, because as you said we inherit it from JS. That could be super confusing. Especially given only one option will work.

I would say if we want to match the mobile SDKs that is fine, but then we should break the inheritance of the options from the JS SDK so we can remove `bufferSize` from the `transportOptions`.

File: CHANGELOG.md [hunk 4]
```diff
@@ -10,13 +14,6 @@
 
 ## 4.8.0
 
-### Various fixes & improvements
-
-- fix(attachments): Maximum call stack exceeded error for large attachments (#2579) by @old
-- fix(event): Message event contains stacktrace if `attachStacktrace` o… (#2577) by @krystofwoldrich
-- chore: update scripts/update-cocoa.sh to 7.29.0 (#2571) by @github-actions
-- chore: update scripts/update-android.sh to 6.6.0 (#2572) by @github-actions
```
Comment (marandaneto) [inline CHANGELOG.md:18]: Why is this removed?

File: src/js/options.ts [hunk 5]
```diff
@@ -121,18 +121,30 @@ export interface BaseReactNativeOptions {
...
+
+  /**
+   * The max queue size for capping the number of envelopes waiting to be sent by Transport.
+   */
+  maxQueueSize?: number;
+}
+
+export interface ReactNativeTransportOptions extends BrowserTransportOptions {
+  /**
+   * @deprecated use `maxQueueSize` in te root of the SDK options.
```
Comment (marandaneto) [inline src/js/options.ts:133]:
```suggestion
   * @deprecated use `maxQueueSize` in the root of the SDK options.
```

File: src/js/options.ts [hunk 6]
```diff
@@ -121,18 +121,30 @@ export interface BaseReactNativeOptions {
...
+   *
+   * Setting this option does not have any effect.
+  */
```
Comment (marandaneto) [inline src/js/options.ts:136]: H: Not sure if this comment makes sense because the code does:

> const maxQueueSize = passedOptions.maxQueueSize
>     // eslint-disable-next-line deprecation/deprecation
>     ?? passedOptions.transportOptions?.bufferSize
>     ?? DEFAULT_OPTIONS.maxQueueSize;

So it does have an effect, right?

File: src/js/options.ts [hunk 7]
```diff
@@ -121,17 +121,29 @@ export interface BaseReactNativeOptions {
...
+   * The max queue size for capping the number of envelopes waiting to be sent by Transport.
+   */
+  maxQueueSize?: number;
+}
+
+export interface ReactNativeTransportOptions extends BrowserTransportOptions {
+  /**
+   * @deprecated use `maxQueueSize` in the root of the SDK options.
+  */
+  bufferSize?: number;
```
Comment (marandaneto) [inline src/js/options.ts:135]: By doing this, if I do:
```
Sentry.init({
	transport(transportOptions) {
	    transportOptions.bufferSize = 100;
	    // ...
	  },
}
```

Do I see in the auto-complete that `bufferSize` is deprecated? 
Should the linter work?
it's not working for me at least.

Comment (krystofwoldrich) [reply src/js/options.ts:135]: I think it is some ts and lint caching problem, it works for me after reloading my VSC window.

Screenshot 2022-11-04 at 14 28 05

Try that I hope it will help.

Comment (marandaneto) [reply src/js/options.ts:135]: No need to block this, just wanted to be sure that it works.
